src/registry/customerProfile/prefill.stories.ts

```diff
@@ -0,0 +1,253 @@
+import type {CustomerProfileComponentSchema} from '@open-formulieren/types';
+import type {Meta, StoryObj} from '@storybook/react-vite';
+import {expect, userEvent, waitFor, within} from 'storybook/test';
+
+import type {FormSettings} from '@/context';
+import {rsValue} from '@/registry/storybook-helpers';
+import {withFormSettingsProvider, withFormik} from '@/sb-decorators';
+
+import {FormioCustomerProfile} from './index';
+
+const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+export default {
+  title: 'Component registry / special / profile / prefill',
+  component: FormioCustomerProfile,
+  decorators: [withFormSettingsProvider, withFormik],
+  args: {
+    componentDefinition: {
+      id: 'customerProfile',
+      type: 'customerProfile',
+      key: 'customerProfile',
+      label: 'Profile',
+      digitalAddressTypes: ['email', 'phoneNumber'],
+      shouldUpdateCustomerData: false,
+    },
+  },
+} satisfies Meta<typeof FormioCustomerProfile>;
+
+type Story = StoryObj<typeof FormioCustomerProfile>;
+
+export const WithOnePrefilledDigitalAddress: Story = {
+  parameters: {
+    formSettings: {
+      componentParameters: {
+        customerProfile: {
+          fetchDigitalAddresses: async () => [
+            {type: 'email', addresses: ['foo@test.com']},
+            {type: 'phoneNumber', addresses: ['0612345678']},
+          ],
+          portalUrl: 'https://example.com',
+        },
+      } satisfies FormSettings['componentParameters'],
+    },
+  },
+  play: async ({canvasElement}) => {
+    // Wait for the prefill to finish and the dropdowns to be rendered
+    await sleep(100);
+
+    const canvas = within(canvasElement);
+    const emailField = canvas.getByLabelText('Email');
+    const phoneNumberField = canvas.getByLabelText('Phone number');
+
+    // Wait for the prefill to finish
+    await waitFor(() => {
+      // Both email and phone number fields are displayed as dropdowns
+      expect(emailField).toHaveRole('combobox');
+      expect(phoneNumberField).toHaveRole('combobox');
+    });
+
+    // The fetched addresses are displayed in the dropdowns
+    expect(rsValue(emailField)).toBe('foo@test.com');
+    expect(rsValue(phoneNumberField)).toBe('0612345678');
```

Comment:
iirc, react-select does have proper label + hidden input linking so you should just be able to do `expect(canvas.getByLabelText(...)).toHaveValue(...)` - is there a reason this doesn't work here?

And in another PR I added an option to render a submit button and you should be able to pass an `onSubmit` parameter, so you I'd much rather prefer we test what the submission data looks like instead of DOM inspection:

```ts
expect(context.parameters.formik.onSubmit).toHaveBeenCalledWith({customerProfile: [...]})
```